modules/test-structure/save_test_data.go

```diff
@@ -282,3 +282,30 @@ func CleanupTestData(t *testing.T, path string) {
 		logger.Logf(t, "%s does not exist. Nothing to cleanup.", path)
 	}
 }
+
+// CleanupTestData cleans up the .test-data folder inside the given folder.
+// If deleteParent is true, CleanupTestDataFolder will also delete the given folder.
+func CleanupTestDataFolder(t *testing.T, path string, deleteParent bool) {
+	if deleteParent {
+		deleteFolder(t, path)
+	} else {
+		testDataPath := filepath.Join(path, ".test-data")
+		deleteFolder(t, testDataPath)
+	}
+}
+
+// cleanupTestDataFolder deletes the given folder.
+func deleteFolder(t *testing.T, path string) {
+	exists, err := files.FileExistsE(path)
+	if err != nil {
+		t.Fatalf("Failed to clean up test data folder at %s: %v", path, err)
+	}
+
+	if exists {
```

Comment:
Do you think the following is more or less readable?

```
if !exists {
    logger.Logf(t, "%s does not exist. Nothing to cleanup", path)
    return
}
if err := os.RemoveAll(path); err != nil {
    t.Fatalf("Failed to clean up test data folder at %s: %v", path, err)
}
```

or even

```
switch exists {
case true:
    if err := os.RemoveAll(path); err != nil {
        t.Fatalf("Failed to clean up test data folder at %s: %v", path, err)
    }
case false:
    logger.Logf(t, "%s does not exist. Nothing to cleanup", path)
}
```

Just trying to show some alternatives. Any of them would work. IMHO, the first example shown above is the cleanest/most idiomatic.